jebaney/caampr-backend
# Creating stub lambda using dagger for DI.  Setting up aws and sam to …

## Notes
* Adds gradle as build manager for the project
* Adds dagger2 as the dependency injection (DI) framework
* Adds a `GetGearList` lambda which ignores any input and just returns a hardcoded list of one piece of gear for testing
* Adds simple stub cloudformation (CFN) template to set up the lambda and IAM role

## Testing
* cd into caampr-backend module
* `gradle clean && gradle build` - succeed and unit tests pass
* `gradle buildAWSLambdaZip` - builds zip in build/distributions/caampr-backend-1.0-SNAPSHOT.jar
* cd scripts
* `./sam-helper package jbaney` -  succeed
* `./sam-helper deploy jbaney` - succeed
* log into AWS console and see the `Caampr-GetGearListjbaney` function
* run test payload and succeeds
* logs show the following repsonse:

```
{
  "statusCode": 200,
  "headers": {},
  "body": "[{\"id\":1,\"name\":\"Tent Stakes\",\"title\":\"Gray Bunny Solid Steel Tent Stakes\",\"description\":\" Premium quality solid iron tent stake kit comes with a set of 12 pegs and a convenient carry bag to take when camping or backpacking.\",\"link\":{\"id\":100,\"url\":\"https://smile.amazon.com/gp/product/B07NS77PDQ\",\"type\":\"AMAZON\"},\"imageUrl\":\"https://images-na.ssl-images-amazon.com/images/I/51y-NmeXHdL._AC_SL1000_.jpg\"}]"
}
```
